src/common/crypto/mbedtls/cryptoprovider.cpp

```diff
@@ -641,6 +642,37 @@ Error MbedTLSCryptoProvider::ASN1DecodeOID(const Array<uint8_t>& inOID, Array<ui
     return crypto::ASN1RemoveTag(inOID, dst, MBEDTLS_ASN1_OID);
 }
 
+RetWithError<uuid::UUID> MbedTLSCryptoProvider::CreateSHA1(const uuid::UUID& space, const String& data)
+{
+    constexpr auto version = 5;
+
+    auto dataBuff = Array<uint8_t>(reinterpret_cast<const uint8_t*>(data.Get()), data.Size());
+
+    StaticArray<uint8_t, cSHA1InputDataSize> buffer = space;
+
+    auto err = buffer.Insert(buffer.end(), dataBuff.begin(), dataBuff.end());
+    if (!err.IsNone()) {
+        return {{}, AOS_ERROR_WRAP(err)};
+    }
+
+    StaticArray<uint8_t, cSHA1DigestSize> sha1;
+
+    sha1.Resize(sha1.MaxSize());
+
+    int ret = mbedtls_sha1(buffer.Get(), buffer.Size(), sha1.Get());
+    if (ret != 0) {
+        return {{}, AOS_ERROR_WRAP(ret)};
+    }
+
+    // copy lowest 16 bytes
+    uuid::UUID result = Array<uint8_t>(sha1.Get(), uuid::cUUIDLen);
+
+    result[6] = (result[6] & 0x0f) | uint8_t((version & 0xf) << 4);
```

Comment:
can you please add more specific comments that explains this magic?

Reply:
done